magenta/music/musicxml_parser.py

```diff
@@ -0,0 +1,678 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""MusicXML parser.
+
+Simple MusicXML parser used to convert MusicXML
+into tensorflow.magenta.NoteSequence.
+"""
+
+# Imports
+# Python 2 uses integer division for integers. Using this gives the Python 3
+# behavior of producing a float when dividing integers
+from __future__ import division
+
+import xml.etree.ElementTree as ET
+from fractions import Fraction
+from zipfile import ZipFile
+from magenta.music import constants
+
+
+class MusicXMLParserState(object):
+  """Maintains internal state of the MusicXML parser"""
+  def __init__(self):
+    # Default to one division per measure
+    self.divisions = 1
+
+    # Default to a tempo of 120 beats per minute
+    # (This is the default tempo according to the
+    # Standard MIDI Files 1.0 Specification)
+    self.tempo = 120
```

Comment:
Ok, I've changed it and the corresponding documentation. I originally called it tempo because in MusicXML it is called "tempo". Since Magenta calls it qpm, I also updated the comment to explain that the two are equivalent.